### PR TITLE
Decorator test for profiler (usage example)

### PR DESCRIPTION
**Pull request by viniciusdc**
This is an attempt to use the new `@profiling` decorator (see #1110 ).

### CODE REVIEW

File: conda_forge_tick/update_prs.py [hunk 1]
```diff
@@ -176,14 +177,38 @@ def close_dirty_prs(gx: nx.DiGraph, dry_run: bool = False) -> nx.DiGraph:
...
-    current_time = now.strftime("%d-%m-%Y") + "_" + now.strftime("%H_%M_%S")
+# profiling decorator
+def profiling():
+    def _profiling(f):
+        @wraps(f)
+        def __profiling(*rgs, **kwargs):
+            prof = cProfile.Profile()
+            prof.enable()
+
+            # out_result = f(*rgs, **kwargs) should we expect an output ?
```
Comment (CJ-Wright) [inline conda_forge_tick/update_prs.py:188]: I would allow for that possibility

File: conda_forge_tick/update_prs.py [hunk 2]
```diff
@@ -176,14 +177,38 @@ def close_dirty_prs(gx: nx.DiGraph, dry_run: bool = False) -> nx.DiGraph:
...
-def main(args: "CLIArgs") -> None:
-    # get current time
-    now = datetime.now()
-    current_time = now.strftime("%d-%m-%Y") + "_" + now.strftime("%H_%M_%S")
+# profiling decorator
+def profiling():
+    def _profiling(f):
+        @wraps(f)
+        def __profiling(*rgs, **kwargs):
+            prof = cProfile.Profile()
```
Comment (CJ-Wright) [inline conda_forge_tick/update_prs.py:185]: You might consider structuring this as a context manager, that way we make certain to clean up even if the underlying function fails

Comment (viniciusdc) [reply conda_forge_tick/update_prs.py:185]: I will do that, thanks for the comment

Comment (viniciusdc) [reply conda_forge_tick/update_prs.py:185]: just reading more about, I don't have much experience with

Comment (viniciusdc) [reply conda_forge_tick/update_prs.py:185]: `cProfile` have a supported Profile class that can be used as an Contex Manager, like this:
```python
with cProfile.Profile() as pr:
    # ... do something ...
```

File: conda_forge_tick/update_prs.py (hunk 1 above)
Comment (viniciusdc) [reply conda_forge_tick/update_prs.py]: done

**Comment by CJ-Wright** [comment]
Should #1111 and #1110 be merged into the same PR?

**Comment by viniciusdc** [comment]
> Should #1111 and #1110 be merged into the same PR?

I didn't thought about that when I created, but you are right. I will set the changes from this one^ to that #1110

**Pull request closed without merging**